traps.py

```diff
@@ -1,36 +1,74 @@
 from memory import mem_read, mem_write, Registers, reg_read, reg_write
-from utils import sign_extend, ushort
+from utils import sign_extend
 from enum import Enum
+from control_unit import Halt
+import sys
 
 
 def _GETC():
     """get character from keyboard"""
-    pass
+    #we need to cast the comming char in 16-bit location
```

Comment:
Don't do imports in this function, this will cause overhead, as this function is called repeatedly during program life time.
take a look at [this example](http://code.activestate.com/recipes/134892-getch-like-unbuffered-character-reading-from-stdin/).